USER_MANUAL.md

```diff
@@ -1,12 +1,355 @@
+-   potential repititions are denoted by triple dots (`...`).
+
+In the end, we require that the file and module hierarchies both form a
+directed acyclic graph (DAG). This is, no file may recursively require itself,
+and likewise, no module may recursively import itself.
+
+We now zoom in further to discuss the various kinds of sentences contained in K
+specifications:
+
+1.  sentences that define our *system's primitives*, including:
+
+    -   **sort declarations:** define new categories of primitive datatypes
+    -   **Backus-Naur Form (BNF) grammar declarations:** define the
+        operators that inhabit our primitive datatypes
+    -   **lexical syntax declarations:** define lexemes/tokens for the
+        lexer/tokenizer
+    -   **syntax associativity declarations:** specify the
+        associativity/grouping of our declared operators
+    -   **syntax priority declarations:** specify the priority of
+        potential ambiguous operators
+
+2.  sentences that define our *system's state*, including:
+
+    -   **configuration declarations:** define labelled, hierarchical records
+        using an nested XML-like syntax
+
+3.  sentences that define our *system's behavior*, including:
+
+    -   **context declarations:** describe how primitives and configurations
+        can simplify
+    -   **context alias declarations:** define templates that can generate new
+        contexts
+    -   **rule declarations:** define how the system transitions from one state
+        to the next
+
+### K Syntax Overview
+
+Here we provide an overview of the top-level K syntax using a K-style BNF
+notation. Note that this is not a full definition; we focus instead on the
+overall structure of the specification. Syntactic categories which are not
+defined are denoted by names surrounded by curly braces (`{}`). In particular,
+we omit the syntax of strings (`{String}`) and all types of identifiers, e.g.,
+`{ModuleId}`, `{Id}`, `{SortId}`, `{CellId}`, etc. We also omit the syntax of
+attributes (`{Attribute}`), a highly flexible mechanism for modifying the
+interpretation of modules and various kinds of sentences.
+
+Due to the highly generic nature of user-defined K syntax, the parsing process
+of K specifications roughly proceeds in three stages, corresponding to the
+three kinds of declarations shown above.
+
+1.  In the first stage, system primitive declarations are gathered and a parser
+    is constructed.
+2.  In the second stage, the system primitive parser is used to parse
+    configuration declarations; in our BNF grammar below, we use `{TermBubble}`
+    to describe things which can be parsed using the primitive syntax.
+3.  Finally, a third stage parser is constructed containing both primitive and
+    configuration syntax is constructed which is used to parse contexts,
+    context aliases, and rules. In our BNF grammar below, we use the words
+    `{Bubble}` and `{BubbleWithRewrites}` to describe parsers which contain the
```

Comment:
this is incorrect; there is only one type of bubble. What may appear inside that bubble is not part of the outer syntax of K; it's part of a separate grammar that is generated from the user-defined syntax.

Reply:
I removed this section from this PR because it needs more polish.